### ISSUE TITLE
(Memory Leak) Strange behavior in MetricNameService.RenameTransaction

### ISSUE THREAD

**Issue by tiagodaraujo** [newrelic/newrelic-dotnet-agent#337]
During a memory leak analysis, I found a list of transactions stored on the `MetricNameService`. The list stored in the `_transactionNames` instance field, increases progressively, even using the SetTransactionName for grouping or IgnoreTransaction API call.

**Description**
`MetricNameService` stores a list of unique transactions in the instance field `_transactionNames`, [here](https://github.com/newrelic/newrelic-dotnet-agent/blob/main/src/Agent/NewRelic/Agent/Core/Metrics/MetricNameService.cs).
The method `IsMetricNameAllowed` stored the transaction but never use the list and return always `true`.
It seems a strange behavior because the `IsMetricNameAllowed` result is used in the method `RenameTransaction`.
Why do you need an `if` in your code, if the operation is always `true`? [here](https://github.com/newrelic/newrelic-dotnet-agent/blob/v8.34.0/src/Agent/NewRelic/Agent/Core/Metrics/MetricNameService.cs#L56).

**Expected Behavior**
I expect for a web request that I call your API methods, NewRelic.SetTransactionName or NewRelic.IgnoreTransaction, the MetricNameService doesn't store the transaction. e.g.

WebRequest: /product/product-seo-description-123.aspx
Call API: NewRelic.SetTransactionName(null, "Product-Detail-Page")
Records:
WebTransaction/ASP/product/product-seo-description-123.aspx
WebTransaction/MVC/Product/Index/{id}
WebTransaction/Custom/Product-Detail-Page
I expect:
WebTransaction/Custom/Product-Detail-Page

WebRequest: /static/image.png
Call API: NewRelic.IgnoreTransaction()
Records:
WebTransaction/ASP/static/image.png
I expect:
No transaction records.

**Note:** In the NewRelic dashboard, Transactions view, all seems right and shows the expected results. Ignored transactions are not showed and custom transactions are shown as expected.

**[Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results**
I don't have a log to be shared here, but I already shared other information internally with the NewRelic support team member.

**Steps to Reproduce**
Just do some traffic and you can see the duplicated transaction names on the `MetricServiceName`. I extracted a memory dump to have access to the object stored.

**Your Environment**
I using the NewRelic agent 8.34 version in a ASP.NET Core application hosted by a Windows Server with IIS.

**Additional context**
snippet API calls: https://gist.github.com/tiagodaraujo/c3d9ac425019b26b29c66e7f01015470

**Comment by nr-ahemsath** [comment]
@tiagodaraujo: Thank you for bringing this to our attention.  You are correct, based on an initial reading, this logic does not make much sense.  At the very least, the name of the `IsTransactionNameAllowed` method is misleading given the current actual behavior of the method.  I'll be looking into this today.

**Comment by nr-ahemsath** [comment]
@tiagodaraujo See the linked pull request.  Again, thanks for bringing this issue to our attention.  This turns out to be obsolete code that should have been removed when the associated functionality was removed.

**Comment by tiagodaraujo** [comment]
Hi @nr-ahemsath, thanks for address this issue quickly. Does the next release have an estimated date?

**Comment by nr-ahemsath** [comment]
We should be ready to do a release sometime next week.

**Comment by tiagodaraujo** [comment]
Hi @nr-ahemsath,
I upgraded the NewRelic.Agent.Api to the latest version. Did some traffic on my application, extracted a memory dump and I see the same MetricNameService transactions in memory. It's weird because I see in your code that you delete them. I confirmed dll version in my application and is the 8.35.0. I restarted my server and sill the same behaviour. Do I need to update something else in the server?
Thank you
